rpcserver.go

```diff
@@ -2202,13 +2209,105 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 		options = append(options, zpay32.CLTVExpiry(uint64(defaultDelta)))
 	}
 
+	// If we were requested to include routing hints in the invoice, then
+	// we'll fetch all of our available private channels and create routing
+	// hints for them.
+	if invoice.Private {
+		openChannels, err := r.server.chanDB.FetchAllChannels()
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch all channels")
+		}
+
+		graph := r.server.chanDB.ChannelGraph()
+
+		numHints := 0
+		for _, channel := range openChannels {
+			// We'll restrict the number of individual route hints
+			// to 20 to avoid creating overly large invoices.
+			if numHints > 20 {
+				break
+			}
+
+			// Since we're only interested in our private channels,
+			// we'll skip public ones.
+			isPublic := channel.ChannelFlags&lnwire.FFAnnounceChannel != 0
+			if isPublic {
+				continue
+			}
+
+			// Make sure the counterparty has enough balance in the
+			// channel for our amount. We do this in order to reduce
+			// payment errors when attempting to use this channel
+			// as a hint.
+			chanPoint := lnwire.NewChanIDFromOutPoint(
+				&channel.FundingOutpoint,
+			)
+			if amtMSat >= channel.LocalCommitment.RemoteBalance {
+				rpcsLog.Debugf("Channel(%v) does not "+
+					"have enough remote balance to be "+
+					"used as a routing hint for invoice",
+					chanPoint)
+				continue
+			}
+
+			// Make sure the channel is active.
+			var linkActive bool
+			link, err := r.server.htlcSwitch.GetLink(chanPoint)
+			if err == nil {
```

Comment:
Hm, I had a comment about this, not sure why it disappeared ;p

Suggestion:
```go
if err != nil {
    log.Errorf("Unable to get link %v, skipping: %v", chanID, err)
   continue
}
if !link.EligibleToForward() {
    log.Debugf("link %v not eligible, skippping", chanID)
    continue
}
```

Reply:
Fixed.